### ISSUE TITLE
I3510 - False positives for missing resources for ec2:CreateTags

### ISSUE THREAD

**Issue by bgardner-noggin** [aws-cloudformation/cfn-lint#4031]
### CloudFormation Lint Version

1.31.1

### What operating system are you using?

Fedora

### Describe the bug

The ec2:CreateTags IAM permission accepts multiple resource types.

The I3510 appears to be enforcing that every potential resource type is listed

To recreate run

cfn-lint -c I <path/to/reproduce-template.yml>

### Expected behavior

No errors from cfn-lint

### Reproduction template

```yaml
Resources:
    Role:
        Type: AWS::IAM::Role
        Properties:
            RoleName: EC2ImageBuilderDistributionCrossAccountRole
            AssumeRolePolicyDocument:
                Version: '2012-10-17'
                Statement:
                  - Effect: Allow
                    Principal: {AWS: [arn:aws:iam::123456789123:root]}
                    Action: ['sts:AssumeRole']
            Path: /
            ManagedPolicyArns:
              - !Sub "arn:${AWS::Partition}:iam::aws:policy/Ec2ImageBuilderCrossAccountDistributionAccess"
            Policies:
                - PolicyName: ec2snapwrite
                  PolicyDocument:
                    Version: '2012-10-17'
                    Statement:
                        - Effect: Allow
                          Action:
                             - ec2:CreateTags
                          Resource: 'arn:aws:ec2:*::snapshot/*'
                          Condition:
                              StringEquals:
                                  "aws:RequestedRegion": !Ref AWS::Region
```
